web/app/tournament/tourny.py

```diff
@@ -169,3 +172,93 @@ def select_player(tournament_id, tournament_player_id):
         # flash("Could not delete player: {{ tp.name }}")
         return redirect(url_for('.players_index',
             tournament_id=tournament_id))
+
+
+@tournament.route('/<tournament_id>/matches/', methods=['GET', 'POST'])
+def matches_index(tournament_id):
```

Comment:
My expectations about a `POST` to `/<tournament_id>/matches` is that it would accept two player_id's and would be used to create one match at a time (say, if you wanted to do a manual pairing).

Instead, how about splitting off to `POST /<tournament_id>/full_pairing`, which takes as input the pairing parameters that Josh mentions below.